sdk/dotnet/Pulumi.Automation/WorkspaceStack.cs

```diff
@@ -616,14 +615,13 @@ await this.Workspace.RunCommandAsync(new[] { "cancel", "--stack", this.Name, "--
         }
 
         private async Task<CommandResult> RunCommandAsync(
-            IEnumerable<string> args,
+            IList<string> args,
             Action<string>? onStandardOutput,
             Action<string>? onStandardError,
             Action<EngineEvent>? onEngineEvent,
             CancellationToken cancellationToken)
         {
-            var argsList = args.ToList();
-            argsList.AddRange(new List<string>() { "--stack", this.Name });
+            args = args.Concat(new[] { "--stack", this.Name }).ToList();
```

Comment:
This looked like a bug as it never passed `argsList`.

Reply:
Note: I also changed `IEnumerable<string> args` to `IList<string> args` everywhere since the enumerable was enumerated multiple times in some places, so it was being treated like a non-enumerable type anyway.

Reply:
@komalali This was added in https://github.com/pulumi/pulumi/pull/6300 - does this change look like a legit bug fix to you?

Reply:
Yes, the change that actually passes in `--stack` to `RunStackCommand` looks good, thanks for catching that. 

I can't speak to the type change for `IEnumerable` to `IList` though :)

Reply:
Resharper correctly warned about the enumerable being enumerated multiple times, so I fixed that too.